python/ray/train/v2/api/config.py

```diff
@@ -90,13 +90,18 @@ class FailureConfig(FailureConfigV1):
     """Configuration related to failure handling of each training run.
 
     Args:
-        max_failures: Tries to recover a run at least this many times.
+        max_failures: Tries to recover a run from training worker errors at least this many times.
             Will recover from the latest checkpoint if present.
             Setting to -1 will lead to infinite recovery retries.
             Setting to 0 will disable retries. Defaults to 0.
+        controller_failure_limit: The maximum number of controller failures to tolerate.
```

Comment:
Let's make this a DeveloperAPI for now while the failure handling logic is still in flux. It might make more sense for us to categorize by "system" / "user" failures and have people configure based on that, but we can discuss that in a follow-up.

```suggestion
        controller_failure_limit: [DeveloperAPI] The maximum number of controller failures to tolerate.
```